src/entrypoints/service_worker.ts

```diff
@@ -177,5 +194,18 @@ self.addEventListener("message", (message) => {
   }
 });
 
+const catchHandler = async (options) => {
+  const dest = options.request.destination;
+
+  if (dest !== "document") {
+    return Response.error();
+  }
```

Comment:
Can we make sure we filter out any url that starts with `/api` or `/static` ? Those we know for sure are not needing our app shell.